# Exercise 1 (Learning to interrupt)

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

# Exercise 2 (Resetting and Restarting the Kernel)

In [2]:
print = 'PRINT'

In [3]:
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


# Exercise 3 (Timing Stuff)

In [1]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [3]:
%%timeit
squares = [i**2 for i in range(10)]

1.01 μs ± 77 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


In [5]:
squares = list()
for i in range(10):
    squares.append(i**2)
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [6]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.29 μs ± 110 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


1.01 μs and 1.29 μs are the respective means of each run. 77 ns and 110 ns are the respective standard deviations for each run. 7 runs means that the code was ran 7 times, and 1,000,000 loops each means that for each run, the code was looped 1,000,000 times.